libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/auth/RustMatrixAuthenticationService.kt

```diff
@@ -251,18 +251,19 @@ class RustMatrixAuthenticationService @Inject constructor(
                     oidcConfiguration = oidcConfiguration,
                     progressListener = progressListener,
                 )
-
-                client.use { rustClient ->
-                    val sessionData = rustClient.session()
+                val sessionData = try {
+                    client.session()
                         .toSessionData(
                             isTokenValid = true,
                             loginType = LoginType.QR,
                             passphrase = pendingPassphrase,
                             sessionPaths = emptySessionPaths,
                         )
-                    sessionStore.storeData(sessionData)
-                    SessionId(sessionData.userId)
+                } finally {
+                    Disposable.destroy(client)
```

Comment:
It may be an overkill, but maybe we could create a general `.use { ... }` extension function? Something like:

```kotlin
inline fun <T : Any?, R> T.use(block: (T) -> R) =
    try {
        block(this)
    } finally {
        try {
            // N.B. our implementation is on the nullable type `Disposable?`.
            Disposable.destroy(this)
        } catch (e: Throwable) {
            // swallow
        }
    }
```

Reply:
Yes, I though about it :)